tslearn/neighbors.py

```diff
@@ -3,11 +3,17 @@
 time series metrics.
 """
 
+import warnings
 import numpy
-from sklearn import neighbors
 from sklearn.neighbors import (KNeighborsClassifier, NearestNeighbors,
-                               KNeighborsRegressor)
-from sklearn.neighbors.base import KNeighborsMixin
+                               KNeighborsRegressor, VALID_METRICS,
+                               NearestNeighbors)
+try:
+    from sklearn.neighbors._base import KNeighborsMixin
+except ImportError:
+    warnings.warn('Detected an older version of sklearn, which will'
+                  'no longer be supported in a future version.')
```

Comment:
I would remove this warning, as it does not provide a lot of useful information for users:
* We use this trick to handle changes in the API of a dependency, not tslearn itself.
* Version 0.22.X of scikit-learn is very recent, and we have no idea when it will be the minimum supported version for tslearn.